### PR TITLE
Added Phone Number and Email Inputs to Tabular Review

### LINKED ISSUE #747: Add Notification Email & Notification Phone Number Input for Tabular Train Flow
**Describe the solution you'd like**
In the Review step of creating a Tabular trainspace, add optional textfields allowing users to input an email and/or phone number in case if the user wants to be notified when training is complete.

**Additional context**
image

Add input fields (preferably MUI textfields for now) to the view in the screenshot above. Make sure to use the existing react-hook-form code to keep track of the current state of the input fields, and that the dynamodb trainspace item correctly contains the correct notification inputs when the user clicks on the train button.

**Setup Instructions (what branch to work off of)**
Run the following commands
```
git checkout nextjs
git pull origin nextjs
git checkout -b tabular-notification-input
```
FYI: If you are not able to immediately run `git checkout nextjs`, make sure you commit your changes in the current branch or run `git stash` and then execute the above commands

### PR DESCRIPTION
**Pull request by Dashrekker**
Added phone number and email inputs for notification.

https://github.com/DSGT-DLP/Deep-Learning-Playground/assets/47958374/dcdb291d-8604-478b-aad3-bc986d474681

image
image

### CODE REVIEW

**Comment by karkir0003** [comment]
@Dashrekker status on this PR? Any blockers?

**Comment by Dashrekker** [comment]
@karkir0003 I've updated the description so you can see what it looks like and what is being updated to the backend.

**Comment by karkir0003** [comment]
hey @Dashrekker , can u have @farisdurrani or @dwu359 review

**Review by farisdurrani — APPROVED** [review]

**Comment by farisdurrani** [comment]
Approved subject to Daniel's approval. Please update your branch and set this PR to not be a draft @Dashrekker

**Comment by karkir0003** [comment]
@farisdurrani updated pr to not be in draft status

File: frontend/playground-frontend/src/features/Train/features/Tabular/components/TabularReviewStep.tsx [hunk 1]
```diff
@@ -27,16 +36,32 @@ const TabularReviewStep = ({
       <Typography>{`Is Default?: ${trainspace.datasetData.isDefaultDataset}`}</Typography>
       <Typography>{`Target Column: ${trainspace.parameterData.targetCol}`}</Typography>
       <Typography>{`Feature Columns: ${trainspace.parameterData.features.join()}`}</Typography>
+      <Controller
+        control={control}
+        name="notificationPhoneNumber"
+        render={({ field: { onChange } }) => (
+          <TextField label="Phone Number" onChange={onChange} />
```
Comment (karkir0003) [inline frontend/playground-frontend/src/features/Train/features/Tabular/components/TabularReviewStep.tsx]: @Dashrekker do you have any validation mechanisms in place to ensure that a phone number that's entered is valid? Similar thing for email? Might be good to validate this before submitting training request

**Review by karkir0003 — CHANGES_REQUESTED** [review]
validation mechanisms for phone number and email?

@farisdurrani thoughts?

@Dashrekker

**Comment by karkir0003** [comment]
@Dashrekker can u show me screen recording of error made in phone number? if users enter invalid email or phone number, we should not let them make train request

thoughts @farisdurrani @dwu359

**Comment by Dashrekker** [comment]
@karkir0003 I added a video of email and phone number validation to the description.

**Review by karkir0003 — APPROVED** [review]

**Comment by karkir0003** [comment]
looks good @. i think u are safe to merge

@farisdurrani confirm?

**Comment by farisdurrani** [comment]
Yes! Go ahead and merge @Dashrekker

**Pull request merged**